vscode/microsoft-kiota/README.md

```diff
@@ -1,58 +1,107 @@
+Select the OpenAPI description you are interested in and you will be presented with the Kiota OpenAPI Explorer containing all the available endpoints 
 
+<img width="482" alt="Kiota OpenAPI explorer" src="https://github.com/microsoft/kiota/blob/865b5f9a3b6aa495f42a397714b09a08bdb99011/vscode/microsoft-kiota/images/samples/endpointSelectionandTheGenerateIcon.png">
 
-## Requirements
+Select the endpoints to include in your API client as above and click the `generate` icon. Kiota extension will display with the options to generate either client, plugin or manifest.
+
+<img width="482" alt="kiota vscode generate options" src="https://github.com/microsoft/kiota/blob/865b5f9a3b6aa495f42a397714b09a08bdb99011/vscode/microsoft-kiota/images/samples/SelectGenerationOption.png">
+
+Finally, the notification bar at the bottom right will display "Generation Completed Successfully" once the generation is done. Click on your vscode file explorer to see the generated outputs in the current workspace if default selection was your output directory choice or navigate to selected folder to see the output.
+
+## Migrating from older use of lock file to a workspace.json
+
+Latest Kiota extension has been improved to using a workspace instead of a lock file to facilitate having multiple clients, plugins, or manifests within the same workspace. The workspace.json file also provides an opportunity to edit, or regenerate the outputs conveniently.
+
+<img width="482" alt="Working with the extension's workspace" src="https://github.com/microsoft/kiota/blob/865b5f9a3b6aa495f42a397714b09a08bdb99011/vscode/microsoft-kiota/images/samples/GenerationMultipleClientsSameWorkspace.png">
+
+If you previously generated a client that had a lock file, kiota prompts you to migrate to using the workspace once you open the previously generated folder containing a lock file. You can choose to migrate immediately or ask to be reminded later.
+
+<img width="482" alt="Notification to Migrate to extension's workspace" src="https://github.com/microsoft/kiota/blob/865b5f9a3b6aa495f42a397714b09a08bdb99011/vscode/microsoft-kiota/images/samples/MigratePromptOnOpeningFolder.png">
+
+You can also access the same feature later by using the contextual option on the lock file, by right-clicking on the lock file and selecting the option to migrate.
+
+<img width="482" alt="Contextual Migrate option on right-clicking on lockfile" src="https://github.com/microsoft/kiota/blob/865b5f9a3b6aa495f42a397714b09a08bdb99011/vscode/microsoft-kiota/images/samples/ContextualMigrateLockToWorkspace.png">
+
+Enjoy the benefits of the workspace once the migration is complete.
 
-None.
 
 ## Extension Settings
 
-None.
+See [src/extensionSettings.ts](https://github.com/microsoft/kiota/blob/865b5f9a3b6aa495f42a397714b09a08bdb99011/vscode/microsoft-kiota/src/extensionSettings.ts) file
```

Comment:
Let's not send our users to a source file with the settings. We should have a doc on this in aka.ms/kiota/docs. For now, can we use this README to just invite users to navigate to the settings and type "Kiota" to see all appropriate settings?